universe/supplyverify/syncer.go

```diff
@@ -0,0 +1,116 @@
+package supplyverify
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/fn"
+	"github.com/lightninglabs/taproot-assets/mssmt"
+	"github.com/lightninglabs/taproot-assets/universe"
+	"github.com/lightninglabs/taproot-assets/universe/supplycommit"
+)
+
+// SupplyLeafFetcher is an interface that can be used to fetch information from
+// a Universe.
+type SupplyLeafFetcher interface {
+	// FetchSupplyLeaves fetches the supply leaves for a specific asset
+	// group within a specified block height range.
+	FetchSupplyLeaves(ctx context.Context,
+		assetSpec asset.Specifier, startBlockHeight,
+		endBlockHeight fn.Option[uint32]) (supplycommit.SupplyLeaves,
+		error)
+
+	// Close closes the fetcher and cleans up any resources.
+	Close() error
+}
+
+// SupplyLeafFetcherFactory is a function type that creates SupplyLeafFetcher
+// instances for a given universe server address.
+type SupplyLeafFetcherFactory func(
+	serverAddr universe.ServerAddr) (SupplyLeafFetcher, error)
+
+// SupplyLeafStore is an interface for storing supply commitment leaves.
+type SupplyLeafStore interface {
+	// ApplySupplyUpdates stores a batch of supply update events to the
+	// database without requiring a transition.
```

Comment:
Without requiring one? As in it'll insert them on disk as supply updates? 

If it's just inserting them as updates, and not trying to construct the final tree (yet), then what's it returning as the node here?